### PR TITLE
feat(ui): fetch account data on demand

### PR DESCRIPTION
**Pull request by userquin**
This PR includes:
- fetch account data when hovering on the account link (on demand)
- ~~add status card skeleton when resolving the account data~~

This will also prevent fetching account data of all accounts in the viewport while scrolling.

We can change the skeleton or switch to use an spinner loader.

Check this video https://streamable.com/4kk87e : added few seconds timeout to see the skeleton on first fetch (afterwards it is immediate)

/cc @patak-dev @shuuji3

### CODE REVIEW

**Comment by patak-dev** [comment]
Is the skeleton still working for you? I can't see it when I test the preview: https://deploy-preview-2632--elk-zone.netlify.app/

**Comment by userquin** [comment]
skeleton logic removed, only fetch on hover

File: components/account/AccountHoverCard.vue [hunk 1]
```diff
@@ -5,7 +5,7 @@ const { account } = defineProps<{
   account: mastodon.v1.Account
 }>()
 
-const relationship = useRelationship(account)
+const relationship = computed(() => account ? useRelationship(account).value : undefined)
```
Comment (patak-dev) [inline components/account/AccountHoverCard.vue:8]: I don't think we should call `useRelationship` inside a `computed`. I think you should extract `useRelationship` to an outside `ref`. If you don't want that ref (that changes once `fetchRelationships()` finished), then you could abstract a helper around `requestedRelationships`

Comment (userquin) [reply components/account/AccountHoverCard.vue]: done

Comment (userquin) [reply components/account/AccountHoverCard.vue]: too many `useRelationship` usages to refactor it here: we should debonce the call

**Review by patak-dev — APPROVED** [review]

**Pull request merged**